### PR TITLE
mgr/deepsea: check for inflight completions when starting event reader, cleanup logging and comments

### PR DESCRIPTION
**Pull request by tserong**
This is a bunch of shuffling comments around, fixing the names of a couple of private variables, and notably handing the possibility of salt-api falling over while we're waiting for some event to complete.

### CODE REVIEW

File: src/pybind/mgr/deepsea/module.py [hunk 1]
```diff
@@ -121,12 +121,16 @@ def available(self):
 
 
     def get_inventory(self, node_filter=None):
+        # Note that this will raise an exception (e.g. if the salt-api is down,
```
Comment (sebastian-philipp) [inline src/pybind/mgr/deepsea/module.py]: Would it make sense to add this as a docstring instead?

File: src/pybind/mgr/deepsea/module.py [hunk 2]
```diff
@@ -307,13 +318,42 @@ def _set_last_failure_msg(self, msg):
...
+                        data = next(iter(return_dict.items()))[1]
+                        self._all_completions[tag]._process_result(data)
+                        # TODO: decide whether it's bad to delete the completion
+                        # here -- would we ever need to resurrect it?
+                        del self._all_completions[tag]
+                    else:
+                        # if the job is not complete, there'll be nothing in the dict
+                        self.log.info("Event {} still pending".format(tag))
+                else:
+                    # TODO: What do we do if the request fails?
```
Comment (sebastian-philipp) [inline src/pybind/mgr/deepsea/module.py]: Maybe throw an exception and let the user know it?

File: src/pybind/mgr/deepsea/module.py [hunk 3]
```diff
@@ -307,13 +318,42 @@ def _set_last_failure_msg(self, msg):
...
+                })
+                if resp.ok:
+                    # jobs.lookup_jid returns a dict keyed by hostname.
+                    return_dict = resp.json()['return'][0]
+                    if return_dict:
+                        # If the job is complete, there'll be one item in the dict.
+                        self.log.info("Event {} complete".format(tag))
+                        # The key is the salt master hostname, but we don't care
+                        # about that, so just grab the data.
+                        data = next(iter(return_dict.items()))[1]
```
Comment (sebastian-philipp) [inline src/pybind/mgr/deepsea/module.py]: `return_dict` always contains only one entry?

File: src/pybind/mgr/deepsea/module.py (hunk 1 above)
Comment (tserong) [reply src/pybind/mgr/deepsea/module.py]: Done

File: src/pybind/mgr/deepsea/module.py (hunk 3 above)
Comment (tserong) [reply src/pybind/mgr/deepsea/module.py]: Yep (at least, if the thing is complete). The entry is the host on which the runner executed.

File: src/pybind/mgr/deepsea/module.py (hunk 2 above)
Comment (tserong) [reply src/pybind/mgr/deepsea/module.py]: The user won't see this directly, as this is happening internally to the module, not in response to user commands. I've rejigged it slightly so it logs a warning, then can keep iterating through any other outstanding completions, then carry on listening to the salt event bus.

Comment (sebastian-philipp) [reply src/pybind/mgr/deepsea/module.py]: We have multiple ways no notify users: one way would be to use the Completion objects to transfer the error, like in `rook.module.RookOrchestrator#wait`. Another one would be to leverage the `progress` module and set the `message` to `str(e)`

Comment (tserong) [reply src/pybind/mgr/deepsea/module.py]: If this request fails, all it means is that we weren't able to query the status of the inflight job. It doesn't tell us if the job succeeded or failed, just that we currently have NFI what's going on with that job.

**Pull request merged**